### PR TITLE
Modal Side Sheet

### PR DESCRIPTION
**Pull request by eskil0312**
Starting with modal side sheet, regular side sheet will be implemented in a new branch

### CODE REVIEW

File: src/components/general/SideSheet/Modal/styles.less [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                .headerIcons{
+                    align-self: flex-end;
+                    display: flex;
+                    flex: 0;
+                }
+            }
+        }
+
+        &.comfortable {
+            --header-font-size: calc(var(--grid-unit) * 3px);
```
Comment (MartinForre) [inline src/components/general/SideSheet/Modal/styles.less]: Are these the only changes between comfy and compact?

File: src/components/general/SideSheet/Modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    const containerClassName = classNames(styles.container, {
+        [styles.show]: isShowing,
+    });
+    const content = useMemo(() => {
+        if (!show) {
+            return null;
+        }
+
+        if (scrollable) {
```
Comment (MartinForre) [inline src/components/general/SideSheet/Modal/index.tsx]: Would this work if it was "scrollable" all the time? Could simplify the code and usage of the component quite a bit

File: src/components/general/SideSheet/Modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                </div>
+            );
+        }
+
+        return <div className={styles.content}>{children}</div>;
+    }, [children, scrollable, show]);
+
+    return (
+        <Overlay show={show}>
+            <div className={containerClassName} onClick={close}>
```
Comment (MartinForre) [inline src/components/general/SideSheet/Modal/index.tsx]: Wasn't there a common Scrim component? 🤔

File: src/components/general/SideSheet/Modal/styles.less [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    opacity: 0;
+    transition: opacity .2s ease;
+    background-color: rgba(0, 0, 0, .32);
+
+    .modalSideSheet {
+        position: fixed;
+        top: 0; bottom: 0;
+        right: 0;
+        width: .grid-unit(80px)[];
+        transform: translateX(640px);
```
Comment (MartinForre) [inline src/components/general/SideSheet/Modal/styles.less]: Use the grid unit here?

File: src/components/general/SideSheet/Modal/styles.less [hunk 5]
```diff
@@ -0,0 +1,94 @@
+@import '~styles/mixins.less';
+@import '~styles/variables.less';
+@import '~styles/colors.less';
```
Comment (MartinForre) [inline src/components/general/SideSheet/Modal/styles.less]: No need to import variables or colors. They're available as global variables within the fusion root component. In fact, they clutter the transpiled code by adding `:root { --grid-unit: 8; // etc. }` a bunch of times to the output and the DOM

File: src/components/general/SideSheet/Modal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                        </div>
+                    </header>
+                    {content}
+                </div>
+            </div>
+        </Overlay>
+    );
+};
+
+const Overlay = ({ children, show }) => {
```
Comment (MartinForre) [inline src/components/general/SideSheet/Modal/index.tsx]: Extract to common component? E.g. `OverlayPortal`

File: src/components/general/SideSheet/Modal/styles.less (hunk 1 above)
Comment (eskil0312) [reply src/components/general/SideSheet/Modal/styles.less]: I think so, but there is no drawings of compact mode so I'll have to discuss with Lotte

File: src/components/general/SideSheet/Modal/index.tsx (hunk 3 above)
Comment (eskil0312) [reply src/components/general/SideSheet/Modal/index.tsx]: There was, was just removed for debugging. Re-added it

File: src/components/general/SideSheet/Modal/styles.less (hunk 1 above)
Comment (eskil0312) [reply src/components/general/SideSheet/Modal/styles.less]: Discussed with Lotte, these are the only changes between comf and comp

**Pull request merged**